src/decanter/core/core_api/predict_input.py

```diff
@@ -49,21 +51,39 @@ def __init__(
         self.pred_body = CoreBody.PredictBody.create(
             data_id='tmp_data_id', model_id='tmp_model_id', callback=callback,
             keep_columns=keep_columns, threshold=threshold, version=version)
+        self.select_by = select_by
+        self.select_model_by = select_model_by
+        self.select_model_id = select_model_id
 
     def getPredictParams(self):
         """Using pred_body to create the JSON request body for prediction.
 
         Returns:
             :obj:`dict`
         """
-        setattr(self.pred_body, 'data_id', self.data.id)
-        setattr(self.pred_body, 'model_id', self.experiment.best_model.id)
+        class_ = self.__class__.__name__
+        if self.select_by == 'by_CVbest':
+            setattr(self.pred_body, 'data_id', self.data.id)
+            setattr(self.pred_body, 'model_id', self.experiment.best_model.id)
+        elif self.select_by == 'by_RC':
+            for rec in self.experiment.recommendations:
+                if self.select_model_by == rec['evaluator']:
+                    setattr(self.pred_body, 'data_id', self.data.id)
+                    setattr(self.pred_body, 'model_id', rec['model_id'])
+        elif self.select_by == 'by_model_id':
+            if self.select_model_id in self.experiment.models:
+                setattr(self.pred_body, 'data_id', self.data.id)
+                setattr(self.pred_body, 'model_id', self.select_model_id)
+            else:
+                logger.error('[%s] no such model ID', class_, self.select_model_id) 
```

Comment:
if user pass wrong model id, how will it predict?

Reply:
if user pass wrong model id or metric not in the model metric list, will raise the error.
and the [data_id and model_id] will become ['tmp_data_id' and 'tmp_model_id']

Reply:
where do you throw the error?